### PR TITLE
Swap out Optim.jl NelderMead.jl

### PR DESCRIPTION
**Pull request by jwscook**
This PR removes Optim.jl as a dependency and replaces it with [NelderMead.jl](https://github.com/jwscook/NelderMead.jl) given the context of https://github.com/jwscook/NelderMead.jl/issues/7.

A point worth noting is that the size of the simplex is taken to be the standard deviation of the starting position (or 1 in the case that the standard deviation is zero). There may be a better simplex size given the range of sensible values of $\alpha_0$; please advise.

Closes https://github.com/jwscook/NelderMead.jl/issues/7.

### CODE REVIEW

File: src/TableTransforms.jl
Comment (juliohm) [inline src/TableTransforms.jl:19]:
```suggestion
using NelderMead: optimise
```

File: src/transforms/projectionpursuit.jl
Comment (juliohm) [inline src/transforms/projectionpursuit.jl:117]:
```suggestion
  res = optimise(f, α₀, iszero(l) ? one(l) : l, xtol_rel=10eps())
  first(res)
```

File: test/Project.toml
Comment (juliohm) [inline test/Project.toml:14]:
```suggestion
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Thank you @jwscook for the PR ❤️ Regarding the initial guess, it is guaranteed to have norm 1. Do you have suggestions for the simplex size in this case?

File: test/Project.toml
Comment (jwscook) [reply test/Project.toml:14]: Woops! Good catch.

**Comment by jwscook** [comment]
> Thank you @jwscook for the PR ❤️ Regarding the initial guess, it is guaranteed to have norm 1. Do you have suggestions for the simplex size in this case?

I'm not entirely sure what the right size should be, but after some trial and error a value of 1/2 seems to work.

File: src/transforms/projectionpursuit.jl
Comment (juliohm) [inline src/transforms/projectionpursuit.jl:115]:
```suggestion
  optimise(f, α₀, 1/2, xtol_rel=10eps()) |> first
```

**Review by juliohm — APPROVED** [review]

**Comment by juliohm** [comment]
That is lovely @jwscook. I will just wait for the tests to pass once again and will then merge it. Did you consider moving NelderMead.jl to JuliaOptim or some other organization with multiple maintainers? It should get more visibility there. :)

**Pull request merged**
